### PR TITLE
Bugfixes for `lmorpho`

### PR DESCRIPTION
**Pull request by halfflat**
* Ignore dendrite branches with negative radii arising from correlated child diameter distribution.
* Fix fencepost errors in morphology discretization.
* Rename `tip.p` to `tip.point`.

### CODE REVIEW

File: lmorpho/morphology.cpp [hunk 1]
```diff
@@ -39,16 +39,19 @@ void section_geometry::segment(double dx) {
     double x = length/nseg;
 
     // Scan segments for sample points.
-    for (unsigned i = 1; i<npoint;) {
+    unsigned i = 1;
+    for (;;) {
         if (right>x) {
             double u = (x-left)/(right-left);
             sampled.push_back(lerp(points[i-1], points[i], u));
```
Comment (w-klijn) [inline lmorpho/morphology.cpp]: What if we have only one point?

File: lmorpho/morphology.cpp [hunk 2]
```diff
@@ -39,16 +39,19 @@ void section_geometry::segment(double dx) {
...
             unsigned k = sampled.size();
-            sampled_length += distance(sampled[k-2], sampled[k]);
+            sampled_length += distance(sampled[k-2], sampled[k-1]);
```
Comment (w-klijn) [inline lmorpho/morphology.cpp]: What if we have only 1 sample? Dunno if possible

File: lmorpho/morphology.cpp (hunk 1 above)
Comment (halfflat) [inline lmorpho/morphology.cpp]: We already have a check for npoint minimum 2.

File: lmorpho/morphology.cpp (hunk 2 above)
Comment (halfflat) [inline lmorpho/morphology.cpp]: We have already pushed one point back on line 30, and another on line 46.

**Pull request merged**